### PR TITLE
Migrate Incomes to TaskPayment

### LINKED ISSUE #4248: Migrate Incomes to TaskPayment
## Rationale
To achieve _Epic Show top-ups and withdraws in the transaction history_ #4172 .

This issue does not cover `DepositPayment`.

### Functional specification
* Migrate `Incomes` to `TaskPayment`. Remove `Income` table.

### User Interface
All rpc that depends on `Incomes` should remain the same API (no changes of CLI nor Electron)

### Technical specification
Additionally fixed difference between migration and model. In migration `TaskPayment.accepted_ts` and `TaskPayment.settled_ts` are `DATETIME NOT NULL`, but in model they're `IntegerField(null=True)`. Either `NOT NULL` should be dropped or type changed to datetime.

## Dependencies
#4197 

### Sub-components
n/a

### Blockers
n/a

### Additional tests
* Update `Income` test
* No additional integration tests required

## QA
_Description of the QA process for the feature_

#### Base test process
_specify one of: ( if unsure, consult @ZmijaWA or @ederenn )_
** QA limited to the single feature (in case of very trivial changes)
** Smoke Test (the limited set of tests designed to ensure that most critical functionality doesn't experience regressions)
** Full Test (full set of tests for the whole application - in case of more extensive application updates, refactoring etc)

### New feature tests
_Add test scenarios for the new feature_

#### Positive Scenario 1
* Step 1
* Step 2
* ...
* Expected results

#### Positive Scenario 2
* ...

#### Negative Scenario 1
* Step 1
* Step 2
* ...
* Some expected problem
* Expected problem handling

### Possible regressions
* is there any functionality that is likely to experience regressions?
* are there any existing application components that should be tested more thoroughly?

### Development
* [ ] Feature implemented
* [ ] Feature unit/integration tests implemented

### Dev QA
* [ ] Basic tests by the developer pass
* [ ] Unit tests pass
* [ ] Golem integration tests pass
* [ ] Concent integration tests pass
* [ ] Concent acceptance tests pass

### QA team
* [ ] Base scenario passes
* [ ] Additional test 1 passes... 
* [ ] ... 

## Issues encountered during QA
_When adding an issue here, please update testing scenarios and QA progress above_
* [ ] link to issue... 
* [ ] ...

### PR DESCRIPTION
**Pull request by jiivan**
fixes: #4248

### CODE REVIEW

File: golem/database/schemas/032_migrate_income_to_task_payment.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            f'0x{db_row["payer_address"]}',
+            db_row['value_received'],
+            db_row['created_date'],
+        ),
+    )
+    wallet_operation_id = cursor.lastrowid
+    cursor.execute(
+        "INSERT INTO taskpayment"
+        " (wallet_operation_id, node, task, subtask,"
+        "  expected_amount, created_date, modified_date, node,"
```
Comment (Krigpl) [inline golem/database/schemas/032_migrate_income_to_task_payment.py]: `node` is twice? Also shouldn't it be set to `sender_node` from the `db_row`?

File: golem/ethereum/incomeskeeper.py [hunk 2]
```diff
@@ -88,25 +92,42 @@ def expect(
...
-            defaults={
-                'payer_address': payer_address,
-                'value': value,
-                'accepted_ts': accepted_ts,
-            },
-        )
-        if inserted:
+
+        try:
+            income = model.TaskPayment.incomes() \
```
Comment (Krigpl) [inline golem/ethereum/incomeskeeper.py]: offline comment

File: golem/model.py [hunk 3]
```diff
@@ -303,7 +306,14 @@ class CURRENCY(msg_dt.StringEnum):
     recipient_address = CharField()
     amount = HexIntegerField()
     currency = StringEnumField(enum_type=CURRENCY)
-    gas_cost = HexIntegerField(null=True)
+    gas_cost = HexIntegerField()
```
Comment (Krigpl) [inline golem/model.py:309]: Why this change? It's somewhat natural that incoming transactions have no cost (as opposed to zero cost).

File: tests/golem/ethereum/test_incomeskeeper.py [hunk 4]
```diff
@@ -108,32 +118,50 @@ def test_received_batch_transfer_closure_time(self):
...
             transaction_id1,
             payer_address,
             value1,
             accepted_ts1,
         )
-        income1 = Income.get(sender_node=sender_node, subtask=subtask_id1)
-        assert transaction_id1[2:] == income1.transaction
-        income2 = Income.get(sender_node=sender_node, subtask=subtask_id2)
-        assert income2.transaction is None
+        income1 = self._get_income(
```
Comment (Krigpl) [inline tests/golem/ethereum/test_incomeskeeper.py]: Maybe introduce `_assert_income(sender_node, subtask_id, transaction_id)` not to repeat the next 4 lines everywhere.

File: golem/model.py (hunk 3 above)
Comment (jiivan) [reply golem/model.py:309]: To unify this with previous migrations. I mean this field is NOT NULL in migrations.

**Review by Krigpl — APPROVED** [review]
